sqlmesh/core/test/definition.py

```diff
@@ -503,10 +503,22 @@ def _add_missing_columns(
         if not all_columns or query.is_star:
             return query
 
-        query_columns = set(query.named_selects)
-        missing_columns = [col for col in all_columns if col not in query_columns]
-        if missing_columns:
-            query.select(*[exp.null().as_(col) for col in missing_columns], copy=False)
+        if isinstance(query, exp.Union):
```

Comment:
I'm not sure this would work with multiple nested unions

Reply:
I'll try to see if it works with nested unions or adjust the logic

Reply:
I feel like we don't need to traverse the Union but instead check the schema of one of its queries and then just put the whole thing into the subquery.

Additionally, I don't understand why do we need to do this at the query level at all and not just add columns to a dataframe we get after evaluating it.

Reply:
> why do we need to do this at the query level at all instead of just adding columns to a dataframe we get after evaluating it.

+1 to this if we can do it, seems simpler to implement using existing logic.

Reply:
Fair points I adapted the logic to use the previous one for filling the nulls and handling the order during the creation of the view